torch/csrc/jit/serialization/export_module.cpp

```diff
@@ -53,21 +53,50 @@ static IValue Table(
   return Tup(std::move(ivalue_entries));
 }
 
-std::string getModulePath(Node* node) {
-  std::string modulePath = node->scopeName();
-  size_t end = modulePath.size();
-  // Here we remove the source range information to make the
-  // module debugging information shorter and cleaner.
-  if (modulePath[end - 1] == '>') {
-    end = modulePath.rfind('<');
-    if (end > 0 && modulePath[end - 1] == '<') {
-      --end;
+std::string getModulePath(Node* node, const std::string& root_scope_string) {
+  if (!node->callstack()) {
+    return root_scope_string + ".forward";
```

Comment:
Any reason for a hard-coded suffix of ".forward"? Could it be the node's scope name, or function schema name?

Reply:
@iseeyuan, I see think this is hardcoded. This is only if you dont have the callstack info available. When you do have callstack it queries `Functions`'s name.